xray_test.go

```diff
@@ -950,3 +962,98 @@ func setStringFlags(flagSet *flag.FlagSet, flags ...string) []string {
+			resolveRepoName: tests.NpmRemoteRepo,
+			cacheRepoName:   tests.NpmRemoteRepo,
+			projectType:     artUtils.Npm,
+		},
+		{
+			testProjectPath: []string{"nuget", "simple-dotnet"},
+			resolveRepoName: tests.NugetRemoteRepo,
+			cacheRepoName:   tests.NugetRemoteRepo,
+			projectType:     artUtils.Dotnet,
+		},
+		{
+			testProjectPath: []string{"yarn", "yarnproject"},
+			resolveRepoName: tests.YarnRemoteRepo,
+			cacheRepoName:   tests.YarnRemoteRepo,
+			projectType:     artUtils.Yarn,
+		},
+	}
+
+	for _, testcase := range testCases {
+		testSingleTechDependencyResolution(t, testcase.testProjectPath, testcase.resolveRepoName, testcase.cacheRepoName, testcase.projectType)
+	}
+}
+
+func testSingleTechDependencyResolution(t *testing.T, testProjectPartialPath []string, resolveRepoName string, cacheRepoName string, projectType artUtils.ProjectType) {
+	log.Info(fmt.Sprintf("--- Starting %s test ---", projectType.String()))
+	tempDirPath, createTempDirCallback := coretests.CreateTempDirWithCallbackAndAssert(t)
+	defer createTempDirCallback()
+	testProjectPath := filepath.Join(append([]string{filepath.FromSlash(tests.GetTestResourcesPath())}, testProjectPartialPath...)...)
+	assert.NoError(t, biutils.CopyDir(testProjectPath, tempDirPath, true, nil))
+	rootDir, err := os.Getwd()
+	require.NoError(t, err)
+	require.NoError(t, os.Chdir(tempDirPath))
+	defer func() {
+		assert.NoError(t, os.Chdir(rootDir))
+	}()
+
+	cleanUpJfrogHome, err := coretests.SetJfrogHome()
+	assert.NoError(t, err)
+	defer cleanUpJfrogHome()
+	server := &config.ServerDetails{
+		Url:            *tests.JfrogUrl,
+		ArtifactoryUrl: *tests.JfrogUrl + tests.ArtifactoryEndpoint,
+		AccessToken:    *tests.JfrogAccessToken,
+		ServerId:       "test",
+	}
+
+	configCmd := coreCmd.NewConfigCommand(coreCmd.AddOrEdit, "test").SetDetails(serverDetails).SetUseBasicAuthOnly(true).SetInteractive(false)
+	assert.NoError(t, configCmd.Run())
+	context := createContext(t, "repo-resolve="+resolveRepoName, "server-id-resolve="+serverDetails.ServerId, "global=false")
+	err = artCmdUtils.CreateBuildConfig(context, projectType)
+	assert.NoError(t, err)
+
+	artifactoryPathToSearch := cacheRepoName + "-cache/*"
+	output := artifactoryCli.RunCliCmdWithOutput(t, "s", artifactoryPathToSearch)
+	// Before the resolution from Artifactory, we verify whether the repository's cache is empty.
+	assert.Equal(t, "[]\n", output)
+
+	// We execute BuildDependencyTree on a project that hasn't been installed. With the Artifactory server and repository configuration, our expectation is that dependencies will be resolved from there
+	auditParams := (&utils.AuditBasicParams{}).SetServerDetails(server).SetDepsRepo(resolveRepoName)
+	err = runBuildDependencyTreeAccordingToTech(t, auditParams, projectType)
+	require.NoError(t, err)
+
+	// Following resolution from Artifactory, we anticipate the repository's cache to contain data.
+	output = artifactoryCli.RunCliCmdWithOutput(t, "s", artifactoryPathToSearch)
```

Comment:
Adding `--fail-no-op` will fail the test if no results were found
```suggestion
	output = artifactoryCli.RunCliCmdWithOutput(t, "s", artifactoryPathToSearch, "--fail-no-op")
```